tests/unit/ct-completion.test.ts

```diff
@@ -0,0 +1,243 @@
+            line: 0,
+          },
+        },
+      },
+    ],
+    data: 0,
+    detail: "",
+    kind: 1,
+    label: "unit",
+  },
+  {
+    data: 1,
+    detail: "",
+    kind: 1,
+    label: "Prototype-Stubs",
+  },
+];
+
+const vmockExpected = [
+  {
+    data: 0,
+    detail: "",
+    kind: 1,
+    label: "unit",
+  },
+  {
+    data: 1,
+    detail: "",
+    kind: 1,
+    label: "Prototype-Stubs",
+  },
+];
+
+const vmockUnitBarExpected = `
+int vmock_unit_bar(::vunit::CallCtx<> vunit_ctx, int z) {
+  // Enable Stub: vmock_unit_bar_enable_disable(vmock_session);
+  // Disable Stub: vmock_unit_bar_enable_disable(vmock_session, false);
+
+  // Insert mock logic here!
+}
+void vmock_unit_bar_enable_disable(vunit::MockSession &vmock_session, bool enable = true) {
+    using vcast_mock_rtype = int ;
+    vcast_mock_rtype (*vcast_fn_ptr)(int)  = &bar;
+    vmock_session.mock <vcast_mock_rtype (*)(int)> ((vcast_mock_rtype (*)(int))vcast_fn_ptr).assign (enable ? &vmock_unit_bar : nullptr);
+}
+// end of mock for: vmock_unit_bar -------------------------------------------------------------------------------------
+
+`;
+
+const vmockUnitBarExpectedComplete = [
+  {
+    data: 0,
+    detail: "",
+    kind: 1,
+    label: vmockUnitBarExpected,
+  },
+];
+
+const vmockUnitExpected = [
+  {
+    data: 0,
+    detail: "",
+    kind: 1,
+    label: "bar",
+  },
+];
+
+const timeout = 30_000; // 30 seconds
+
+describe("Testing pythonUtilities (valid)", () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+
+  test(
+    "validate coded test completion for lines of interest (// vmock) and extra text. ",
+    async () => {
+      // To achieve 100% coverage
+      // --> Need to mock getChoiceDataFromPython, since "extraText" is not implemented yet
+      const pythonUtilities = await import("../../server/pythonUtilities");
+      vi.spyOn(pythonUtilities, "getChoiceDataFromPython").mockReturnValue({
+        choiceKind: "File",
+        choiceList: ["unit", "Prototype-Stubs"],
+        extraText: "some extra data",
+        messages: [],
+      });
+
+      const tstText = unitTst;
+      const lineToComplete = "// vmock";
+
+      const completionPosition = getCompletionPositionForLine(
+        lineToComplete,
+        tstText
+      );
+
+      // Flag that we are dealing with coded tests
+      const cppTestFlag = { cppTest: true, lineSoFar: lineToComplete };
+      const triggerCharacter = lineToComplete.at(-1);
+
+      const codedTestCompletionData = generateCompletionData(
+        tstText,
+        completionPosition,
+        triggerCharacter,
+        cppTestFlag
+      );
+
+      expect(codedTestCompletionData).toEqual(extraTextMockExpected);
+    },
+    timeout
+  );
+
+  test(
+    "validate coded test completion for lines of interest (// vmock) and without extra text. ",
+    async () => {
+      // No mock for this test to let it excercise the actual python but without extra text
+      const tstText = unitTst;
```

Comment:
Should we make this all be a function with args such as `lineToComplete` + `expectedReturn`? I think all of these (non-mocked) tests now have the same body, but with different inputs/expected, so I would refactor them out into a function that can be called by the tests

Reply:
Refactored code in 805ebe742f3dfb6d82093c3d9606af7a90511b9f .